packages/hooks/src/useSilentLoginAndLogout.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+import Cookies from 'js-cookie';
+
+import { requestOidcAuthentication } from '@deriv-com/auth-client';
+
+/**
+ * Handles silent login and single logout logic for OAuth2.
+ *
+ * @param {{
+ *   is_client_store_initialized: boolean; // Whether the client store has been initialized
+ *   isOAuth2Enabled: boolean; // Whether OAuth2 feature is enabled
+ *   oAuthLogout: () => Promise<void>; // Function to handle OAuth2 logout
+ * }} params - The arguments required for silent login and logout management
+ */
+const useSilentLoginAndLogout = ({
+    is_client_store_initialized,
+    isOAuth2Enabled,
+    oAuthLogout,
+}: {
+    is_client_store_initialized: boolean;
+    isOAuth2Enabled: boolean;
+    oAuthLogout: () => Promise<void>;
+}) => {
+    const loggedState = Cookies.get('logged_state');
+
+    const clientAccounts = JSON.parse(localStorage.getItem('client.accounts') || '{}');
+    const isClientAccountsPopulated = Object.keys(clientAccounts).length > 0;
+
+    useEffect(() => {
+        if (
+            loggedState === 'true' &&
+            !isClientAccountsPopulated &&
+            isOAuth2Enabled &&
+            is_client_store_initialized &&
+            window.location.pathname !== '/callback'
+        ) {
+            // Perform silent login
+            requestOidcAuthentication({
+                redirectCallbackUri: `${window.location.origin}/callback`,
+            });
+        }
+
+        if (loggedState === 'false' && is_client_store_initialized && isOAuth2Enabled && isClientAccountsPopulated) {
```

Comment:
[M]
```suggestion
        if (loggedState !== 'true' && is_client_store_initialized && isOAuth2Enabled && isClientAccountsPopulated) {
```

Reply:
As discussed, there is 3 scenarios. Just true, false and if the logged_state is not there, dont do anything